apps/web/src/middlewares/client-id-middleware.ts

```diff
@@ -0,0 +1,19 @@
+import { NextFetchEvent, NextResponse } from 'next/server'
+import { v4 } from 'uuid'
+import { ExtendedNextReq, MiddlewareFactory, NextMiddleware } from './types'
+
+export const withClientId: MiddlewareFactory = (next: NextMiddleware) => {
+  return async (request: ExtendedNextReq, _next: NextFetchEvent) => {
+    const response = (await next(request, _next)) || NextResponse.next()
+
+    const clientIdCookie = request.cookies.get('client-id')
```

Comment:
btw let's use underscore lowercase for cookie name. It's more commonly used `client_id`